Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/Update.ts

```diff
@@ -79,15 +80,44 @@ export async function updateDocumentById(
       await writeLockReferencedDocuments(mongoCollection, document.outboundRefs, session);
 
       // Perform the document update
-      Logger.debug(`${moduleName}.updateDocumentById: Updating document id ${id}`, traceId);
+      Logger.debug(`${moduleName}.updateDocumentById: Updating document uuid ${documentUuid}`, traceId);
 
-      const { acknowledged, matchedCount } = await mongoCollection.replaceOne({ _id: id }, document, { session });
-      if (acknowledged) {
-        updateResult.response = matchedCount > 0 ? 'UPDATE_SUCCESS' : 'UPDATE_FAILURE_NOT_EXISTS';
+      // if resourceInfo.allowIdentityUpdates is true, the process deletes de old document and inserts a new document.
+      if (resourceInfo.allowIdentityUpdates) {
+        const deleteResult = await deleteDocumentById(
+          { documentUuid, meadowlarkId, resourceInfo, security, validate: true, traceId },
```

Comment:
This function is at too high a level to use here. It manages its own transaction, so the `delete` transaction will be outside the `update` transaction.

Reply:
If in this option I delete a document, but I can't insert the new version, since the upsert and delete are in different transactions, should I reinsert the original document?